src/components/clock/clock.ts

```diff
@@ -0,0 +1,31 @@
+import { sendPacket } from './../../hid/keyboard-hid';
+import { ClockPacket } from '../../hid/packet/clock-packet';
+
+let clockIntervalId: any;
+
+export function initTaskSendClock() {
+    clockIntervalId = setInterval(eventSendClock, 1000);
+}
+export function endTaskSendClock() {
+    clearInterval(clockIntervalId);
+}
+
+function eventSendClock() {
+    let time = getCurrentTimeFormatted();
+    sendPacket(new ClockPacket(time).toPacket());
+}
+
+function getCurrentTimeFormatted(): string {
+    const zeroPad = (num: number, places: number): string => String(num).padStart(places, '0');
```

Comment:
get rid of - shane

Reply:
oh i remember why this is here, we need to pad the string with leading zeroes for minutes/seconds if the number is a single digit (ex 1:9:8pm -> 1:09:08pm)